src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -696,7 +696,7 @@ private Exception coldStartIfNoModel(AtomicReference<Exception> failure, Anomaly
         }
         LOG.info("Trigger cold start for {}", detector.getDetectorId());
         coldStart(detector);
-        return previousException.orElse(new InternalFailure(adID, NO_MODEL_ERR_MSG));
+        return previousException.orElse(exp);
```

Comment:
To catch the resource not found exception on the job scheduler and delay updating real time cache by 1 minute. Otherwise, we only get internal failure and has to process its error msgs.